infra/Pos/Slotting/MemState/Holder.hs

```diff
@@ -21,26 +21,31 @@ import qualified Ether
 import           Pos.Core.Types               (Timestamp)
 
 import           Pos.Slotting.MemState.Class  (MonadSlotsData (..))
-import           Pos.Slotting.Types           (SlottingData (sdPenultEpoch))
+import           Pos.Slotting.Types           (SlottingData,
+                                               addEpochSlottingData,
+                                               getLastEpochIndex,
+                                               getPenultEpochIndex,
+                                               lookupEpochSlottingData)
 
 ----------------------------------------------------------------------------
 -- Transformer
 ----------------------------------------------------------------------------
 
 -- | System start and slotting data
-type SlottingVar = (Timestamp, TVar SlottingData)
+type SystemStart = Timestamp
```

Comment:
Even better is to wrap it into `newtype`, but it is not necessary for this task of course.